packages/components/src/components/navigation/utils.ts

```diff
@@ -33,13 +33,17 @@ export function getHref(url : AppURL | string) : string {
     return typeof url == 'string' ? url : url.toHref();
 }
 
-export function createApplicationUrl(urlProductId: string, currentProductId: string, params: { [key: string]: any }, ...parts) : string | AppURL {
+export function createApplicationUrlFromParts(urlProductId: string, currentProductId: string, params: { [key: string]: any }, ...parts) : string | AppURL {
```

Comment:
I know this is just building off of what I did, but I wonder if we should rename this to be `createProductUrlFromParts` to avoid confusion with `AppURL` since this returns either an `AppURL` or a `string`.

Reply:
yeah, that makes sense to me. I'll rename.